tests/test_serialization.py

```diff
@@ -337,6 +338,47 @@ def test_s3_lazy_load(self):
         check_inference(self, deserialized, model_name, default_device)
         deserialized.close()
 
+    def test_redis(self):
+        test_redis = TestRedis()
+        test_redis.setUpClass()
+        test_redis.setUp()
+        redis_model_path = (
+            f"redis://{test_redis.redis_host}:{test_redis.redis_port}"
+            f"/{model_name}"
+        )
+
+        deserialized_s3 = TensorDeserializer(
+            f"s3://tensorized/{model_name}/model.tensors",
+            device=default_device,
+        )
+        deserialized_s3.to_redis(test_redis.redis_client, model_name)
+        deserialized_s3.close()
+
+        with self.subTest(msg="Testing redis deserialization in eager mode"):
+            deserialized_redis = TensorDeserializer(
+                redis_model_path,
+                device=default_device,
+            )
+            check_deserialized(self, deserialized_redis, model_name)
+            check_inference(
+                self, deserialized_redis, model_name, default_device
+            )
+            deserialized_redis.close()
+
+        with self.subTest(msg="Testing redis deserialization in lazy mode"):
+            deserialized_redis = TensorDeserializer(
+                redis_model_path,
+                device=default_device,
+                lazy_load=True,
+            )
+            check_deserialized(self, deserialized_redis, model_name)
+            check_inference(
+                self, deserialized_redis, model_name, default_device
+            )
+            deserialized_redis.close()
+
+        test_redis.tearDown()
```

Comment:
This would be good to put into a `finally:` clause in case any of the preceding subtests fail